.github/workflows/build-npm-and-release.yml

```diff
@@ -0,0 +1,173 @@
+          docker run --rm -v $(pwd):/build -w /build builder yarn --ignore-scripts 
+          docker run --rm -v $(pwd):/build -w /build builder yarn build:linux
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: index.linux-ia32-gnu.node
+          path: crates/matrix-sdk-crypto-nodejs/lib/index.linux-ia32-gnu.node
+          if-no-files-found: error
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: index.linux-x64-gnu.node
+          path: crates/matrix-sdk-crypto-nodejs/lib/index.linux-x64-gnu.node
+          if-no-files-found: error
+
+  x86_64-unknown-linux-musl:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Enter folder
+        run: |
+          ls -l
+          cd crates/matrix-sdk-crypto-nodejs/
+          ls -l
+
+          docker pull ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-alpine
+          docker tag ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-alpine builder
+
+          docker run --rm -v $(pwd):/build -w /build builder yarn --ignore-scripts 
+          docker run --rm -v $(pwd):/build -w /build builder yarn build:linux:x64:musl
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: index.linux-x64-musl.node
+          path: crates/matrix-sdk-crypto-nodejs/lib/index.linux-x64-musl.node
+          if-no-files-found: error
+
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    needs:
+      - x86_64-apple-darwin
+      - aarch64-apple-darwin
+      - x86_64-unknown-linux-gnu
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          check-latest: true
+      - name: Download all artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: Make npm package
+        run: |
+          echo "Bringing in artifacts"
+          mkdir crates/matrix-sdk-crypto-nodejs/lib
+          ls -l artifacts
+          mv artifacts/* crates/matrix-sdk-crypto-nodejs/lib/
+          
+          echo "Installing npm dependencies"
+          cd crates/matrix-sdk-crypto-nodejs/
+          yarn install --ignore-scripts --frozen-lockfile --registry https://registry.npmjs.org --network-timeout 300000
+          
+          echo "Compiling ts to js for releasing"
+          yarn build:ts
+          ls -l lib
+ 
+          if git log -1 --pretty=%B | grep "^[0-9]\+\.[0-9]\+\.[0-9]\+$";
+          then
+            echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+            npm publish --access public
+          elif git log -1 --pretty=%B | grep "^[0-9]\+\.[0-9]\+\.[0-9]\+";
+          then
+            echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+            npm publish --tag next --access public
```

Comment:
Pulled this bit from the napi example project. So probably wouldn't need this bit